muikku-school-data-pyramus/src/main/java/fi/otavanopisto/muikku/plugins/schooldatapyramus/PyramusUpdater.java

```diff
@@ -515,8 +534,32 @@ public int updateCourseStaffMembers(Long courseId) {
         if (workspaceUserEntity == null) {
           fireCourseStaffMemberDiscovered(staffMember);
         }
+        else {
+          for (int i = currentStaffMembers.size() - 1; i >= 0; i--) {
+            if (currentStaffMembers.get(i).getId().equals(workspaceUserEntity.getId())) {
+              currentStaffMembers.remove(i);
+            }
+          }
+        }
       }
     }
+    
+    // At this point current staff members only contain those that Pyramus no longer acknowledges.
+    // We have somehow missed their original removal event but fake them here to get staff in sync.
+    
+    for (WorkspaceUserEntity currentStaffMember : currentStaffMembers) {
+      try {
+        String courseStaffMemberIdentifier = currentStaffMember.getIdentifier();
+        String staffMemberIdentifier = currentStaffMember.getUserSchoolDataIdentifier().getUserEntity().getDefaultIdentifier();
+        Long courseStaffMemberId = NumberUtils.toLong(StringUtils.substringAfterLast(courseStaffMemberIdentifier, "-"));
```

Comment:
Use PyramusIdentifierMapper instead of explicit Long conversion (getPyramusCourseStaffId and getPyramusStaffId below [check])